sqlmesh/dbt/context.py

```diff
@@ -238,9 +238,13 @@ def context_for_dependencies(self, dependencies: Dependencies) -> DbtContext:
             else:
                 raise ConfigError(f"Source '{source}' was not found.")
 
+        variable_deps = set(dependencies.variables)
```

Comment:
and here you convert back from list to set, maybe these should all just be sets?